pallets/did/src/traits.rs

```diff
@@ -0,0 +1,68 @@
+// KILT Blockchain – https://botlabs.org
+// Copyright (C) 2019-2023 BOTLabs GmbH
+
+// The KILT Blockchain is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The KILT Blockchain is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+// If you feel like getting in touch with us, you can do so at info@botlabs.org
+
+use codec::{Decode, Encode};
+use frame_support::{weights::Weight, RuntimeDebug};
+use scale_info::TypeInfo;
+use sp_runtime::DispatchError;
+use xcm::v2::MultiLocation;
+
+use crate::{did_details::DidDetails, AccountIdOf, Config, DidIdentifierOf};
+
+pub trait DidDocumentHasher<DidIdentifier, DidDetails, Output> {
+	const MAX_WEIGHT: Weight;
+
+	fn calculate_root(did: &DidIdentifier, details: &DidDetails) -> Result<(Output, Weight), DispatchError>;
+}
+
+impl<T: Config> DidDocumentHasher<DidIdentifierOf<T>, DidDetails<T>, T::Hash> for () {
```

Comment:
Just a minor comment. Maybe we could start to use structs instead of `()`. THis would make sure that it's clear what the implementation does.

Reply:
Wasn't kind of known that a trait implemented by `()` does nothing? But sure, I will change that.